src/Compilers/CSharp/Portable/Binder/SwitchBinder_BindPatternSwitch.cs

```diff
@@ -63,67 +198,402 @@ private BoundPatternSwitchSection BindPatternSwitchSection(BoundExpression bound
         }
 
-        private static BoundPatternSwitchLabel BindPatternSwitchSectionLabel(Binder sectionBinder, BoundExpression boundSwitchExpression, SwitchLabelSyntax node, ref DefaultSwitchLabelSyntax defaultLabel, DiagnosticBag diagnostics)
+        private BoundPatternSwitchLabel BindPatternSwitchSectionLabel(
+            Binder sectionBinder, BoundExpression boundSwitchExpression, bool? switchExpressionIsNull, SwitchLabelSyntax node, LabelSymbol label, ref DefaultSwitchLabelSyntax defaultLabel, DiagnosticBag diagnostics)
         {
             switch (node.Kind())
             {
-                case SyntaxKind.CasePatternSwitchLabel:
-                    {
-                        var matchLabelSyntax = (CasePatternSwitchLabelSyntax)node;
-                        var pattern = sectionBinder.BindPattern(
-                            matchLabelSyntax.Pattern, boundSwitchExpression, boundSwitchExpression.Type, node.HasErrors, diagnostics, wasSwitchCase: true);
-                        return new BoundPatternSwitchLabel(node, pattern,
-                            matchLabelSyntax.WhenClause != null ? sectionBinder.BindBooleanExpression(matchLabelSyntax.WhenClause.Condition, diagnostics) : null, node.HasErrors);
-                    }
-
                 case SyntaxKind.CaseSwitchLabel:
                     {
                         var caseLabelSyntax = (CaseSwitchLabelSyntax)node;
                         bool wasExpression;
                         var pattern = sectionBinder.BindConstantPattern(
-                            node, boundSwitchExpression, boundSwitchExpression.Type, caseLabelSyntax.Value, node.HasErrors, diagnostics, out wasExpression, wasSwitchCase: true);
-                        return new BoundPatternSwitchLabel(node, pattern, null, node.HasErrors);
+                            node, boundSwitchExpression, switchExpressionIsNull, boundSwitchExpression.Type, caseLabelSyntax.Value, node.HasErrors, diagnostics, out wasExpression, wasSwitchCase: true);
+                        bool hasErrors = pattern.HasErrors;
+                        if (!hasErrors && pattern.ConstantValue != null && this.FindMatchingSwitchCaseLabel(pattern.ConstantValue) != label)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_DuplicateCaseLabel, node.Location, pattern.ConstantValue?.Value ?? label.Name);
+                            hasErrors = true;
+                        }
+                        return new BoundPatternSwitchLabel(node, label, pattern, null, hasErrors);
                     }
 
                 case SyntaxKind.DefaultSwitchLabel:
                     {
                         var defaultLabelSyntax = (DefaultSwitchLabelSyntax)node;
-                        var pattern = new BoundWildcardPattern(node);
+                        var pattern = new BoundWildcardPattern(node, knownMatchResult: true);
```

Comment:
No. It represents the source syntax (the default label).

---
In reply to: [64956733](https://github.com/dotnet/roslyn/pull/11485#discussion_r64956733) [](ancestors = 64956733)